File: ProjecIntegration.Api/projectintegrationFront/src/app/class/complexe.ts
import { Catalogue } from "./catalogue";
import { SalledeTheatre } from "./salledetheatre";

export class Complexe {
   Name: string;
  Description: string;
  SalleDeTheatres: SalledeTheatre[];
  Catalogues: Catalogue[];

  constructor(
    Name: string,
    Description: string,
    SalleDeTheatres: SalledeTheatre[],
    Catalogues: Catalogue[]
  ) {
    this.Name = Name;
    this.Description = Description;
    this.SalleDeTheatres = SalleDeTheatres;
    this.Catalogues = Catalogues;
  }
}

File: ProjecIntegration.Api/Domain/Entity/TheatherEntity/Complexe.cs
namespace Domain.Entity.TheatherEntity
{
    public class Complexe : BaseEntity
    {

        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Adress { get; set; }=string.Empty;
        public List<SalleDeTheatre>? SalleDeTheatres { get; set; }
        public List<Catalogue>? Catalogue { get; set; }
        //public Entreprise Entreprise { get; set; }
        //public int EntrepriseID { get; set; }
    }
}
